scripts/imgtool/image.py

```diff
@@ -246,14 +258,22 @@ def ecies_p256_hkdf(self, enckey, plainkey):
     def create(self, key, enckey, dependencies=None):
         self.enckey = enckey
 
-        if dependencies is None:
-            dependencies_num = 0
-            protected_tlv_size = 0
-        else:
-            # Size of a Dependency TLV = Header ('BBH') + Payload('IBBHI')
-            # = 16 Bytes
+        protected_tlv_size = 0
+
+        if self.security_counter is not None:
+            # TLV info header + security counter TLV
+            # Size of the security counter TLV: header ('HH') + payload ('I')
+            #                                   = 4 + 4 = 8 Bytes
+            protected_tlv_size = TLV_INFO_SIZE + TLV_SIZE + 4
+
+        if dependencies is not None:
+            if protected_tlv_size == 0:
+                # TLV info header
+                protected_tlv_size = TLV_INFO_SIZE
+            # Size of a Dependency TLV = Header ('HH') + Payload('IBBHI')
+            # = 4 + 12 = 16 Bytes
             dependencies_num = len(dependencies[DEP_IMAGES_KEY])
-            protected_tlv_size = (dependencies_num * 16) + TLV_INFO_SIZE
+            protected_tlv_size += (dependencies_num * 16)
 
```

Comment:
Following my previous comments, you could add

```
if protected_tlv_size != 0:
￼                # add TLV info header
￼                protected_tlv_size += TLV_INFO_SIZE
```

here; at least I think this should work!

Reply:
Done.